fenicsprecice/adapter_core.py

```diff
@@ -95,7 +95,7 @@ def determine_function_type(input_obj):
     if isinstance(input_obj, FunctionSpace):  # scalar-valued functions have rank 0 is FEniCS
         if input_obj.num_sub_spaces() == 0:
             return FunctionType.SCALAR
-        elif input_obj.num_sub_spaces() == 2:
+        elif input_obj.num_sub_spaces() == 2 or input_obj.num_sub_spaces() == 3:
```

Comment:
```suggestion
        elif input_obj.num_sub_spaces() == 2:
            assert(dimensions_of_function_space == 2)  // needs to be implemented
            return FunctionType.VECTOR
        elif input_obj.num_sub_spaces() == 3:
            assert(dimensions_of_function_space == 3)  // needs to be implemented
            return FunctionType.VECTOR            
```
You could additionally check that the dimensions of the function space are matching. In 3D we only accept 3D vectors, in 2D only 2D vectors. Currently in 3D a 2D vector would be accepted here.